### ISSUE TITLE
If there are two read coroutines, an exception is thrown

### ISSUE THREAD

**Issue by csrftoken** [tortoise/tortoise-orm#110]
If there are two read coroutines, an exception is thrown
![image](https://user-images.githubusercontent.com/29940756/54574702-4cdbcb00-4a2c-11e9-998a-5072ad06d1d3.png)

**Comment by csrftoken** [comment]
I create mysql poll, that solved the problem, but I don't think it's the most effective solution

**Comment by grigi** [comment]
Hi, could you drop a small program that repeats the issue, then I'll add it to the test suite and get it fixed.

**Comment by csrftoken** [comment]
This problem will not occur if you use mysql connection pooling.
I saw the new branch of your new_pool and wrote the code for the connection pool function. When will it be available for the main branch.

**Comment by abondar** [comment]
@grigi I think that test should do the trick
```
async def test_concurrent(self):
    await Tournament.create(name='Test')
    await asyncio.gather(*[
        Tournament.first(),
        Tournament.first(),
        Tournament.first(),
    ])
```

**Comment by grigi** [comment]
@csrftoken Due to this being an concurrent ORM, even with a single connection it should be able to handle concurrency if not parallelism (which is what multiple pools should solve). 

Thanks for the test case Andrey, I'll have a go at seeing if that can be solved without connection pooling, as we may end up with the same issue once connections run out, or for DB's which have no parallelism (such as SQLite)

I'll see about getting to it soon.

**Comment by croshchupkin** [comment]
At a glance, no errors seem to happen to sqlite in a similar situation. The code that I tried is:
```
# models.py
from tortoise import Model, fields


class Contact(Model):
    id = fields.IntField(pk=True)
    phone_no = fields.CharField(30, null=False, default='')
    user = fields.ForeignKeyField('models.User', related_name='contacts',
                                  on_delete=fields.CASCADE, null=False)


class User(Model):
    id = fields.IntField(pk=True)
    name = fields.CharField(128, null=False, required=True)
```
The script:
```
import asyncio

from tortoise import run_async, Tortoise

from models import Contact


async def run():
    await Tortoise.init(
        db_url='sqlite://db.sqlite3',
        modules={'models': ['models']}
    )
    await Tortoise.generate_schemas()

    await asyncio.gather(Contact.first(), Contact.first(), Contact.first())


if __name__ == '__main__':
    run_async(run())
```
Please correct me if I missed something, I'm new to async.

**Comment by csrftoken** [comment]
@grigi I hope you'll find out as soon as I'm ready to apply it to production and get a feel for your code.
I can learn a lot from it.

Thanks.

**Comment by grigi** [comment]
@csrftoken SQLite connects via a asyncio-to-thread buffer, and hence all access to the concept was already protected. This was not the case for the other DB's.
I added mutexes around connection aquisition that resolved it for me in my test cases.
Fix in #111 

I'll release 0.11.5 soon.
